### PR TITLE
feat: rewrite e2e test(upstream_chash_query_string_arg_xxx_test) with ginkgo

### PR DESCRIPTION
**Pull request by Jaycean**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues
#1500

___
### New feature or improvement
-  Rewrite e2e test(upstream_chash_query_string_arg_xxx_test) with ginkgo

### CODE REVIEW

**Review by nic-chen — APPROVED** [review]

File: api/test/e2enew/upstream/upstream_chash_query_string_arg_xxx_test.go [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+		time.Sleep(time.Duration(500) * time.Millisecond)
+		basepath := base.APISIXHost
+		res := map[string]int{}
+		for i := 0; i < 180; i++ {
+			url := basepath + "/server_port?var=2&var2=" + strconv.Itoa(i)
+			req, err := http.NewRequest("GET", url, nil)
+			assert.Nil(t, err)
+			resp, err := http.DefaultClient.Do(req)
+			assert.Nil(t, err)
+			respBody, err := ioutil.ReadAll(resp.Body)
```
Comment (starsz) [inline api/test/e2enew/upstream/upstream_chash_query_string_arg_xxx_test.go:95]: Need to assert the err

File: api/test/e2enew/upstream/upstream_chash_query_string_arg_xxx_test.go [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+		t := ginkgo.GinkgoT()
+		time.Sleep(time.Duration(500) * time.Millisecond)
+		basepath := base.APISIXHost
+		res := map[string]int{}
+		for i := 0; i < 180; i++ {
+			url := basepath + "/server_port?var=2&var2=" + strconv.Itoa(i)
+			req, err := http.NewRequest("GET", url, nil)
+			assert.Nil(t, err)
+			resp, err := http.DefaultClient.Do(req)
+			assert.Nil(t, err)
```
Comment (starsz) [inline api/test/e2enew/upstream/upstream_chash_query_string_arg_xxx_test.go:94]: Defer closing the resp.Body at here.

File: api/test/e2enew/upstream/upstream_chash_query_string_arg_xxx_test.go [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+			respBody, err := ioutil.ReadAll(resp.Body)
+			body := string(respBody)
+			if _, ok := res[body]; !ok {
+				res[body] = 1
+			} else {
+				res[body]++
+			}
+			resp.Body.Close()
```
Comment (starsz) [inline api/test/e2enew/upstream/upstream_chash_query_string_arg_xxx_test.go:189]: Ditto.

File: api/test/e2enew/upstream/upstream_chash_query_string_arg_xxx_test.go (hunk 1 above)
Comment (Jaycean) [reply api/test/e2enew/upstream/upstream_chash_query_string_arg_xxx_test.go:95]: done.

File: api/test/e2enew/upstream/upstream_chash_query_string_arg_xxx_test.go (hunk 2 above)
Comment (Jaycean) [reply api/test/e2enew/upstream/upstream_chash_query_string_arg_xxx_test.go:94]: done. It's my negligence. I'll check the previous use cases and correct the same problems

File: api/test/e2enew/upstream/upstream_chash_query_string_arg_xxx_test.go (hunk 3 above)
Comment (Jaycean) [reply api/test/e2enew/upstream/upstream_chash_query_string_arg_xxx_test.go]: done.

**Comment by Jaycean** [comment]
cc @nic-chen @starsz  There seems to be something wrong with CI?

**Comment by nic-chen** [comment]
> cc @nic-chen @starsz There seems to be something wrong with CI?

need to wait for it..

**Comment by starsz** [comment]
> cc @nic-chen @starsz There seems to be something wrong with CI?

Maybe Github is too busy.

**Comment by Jaycean** [comment]
cc @starsz PTAL, Thks.

**Review by starsz — APPROVED** [review]
LGTM.

**Pull request merged**